hub-saml/src/main/java/uk/gov/ida/saml/hub/domain/AuthenticationStatusFactory.java

```diff
@@ -0,0 +1,7 @@
+package uk.gov.ida.saml.hub.domain;
+
+import uk.gov.ida.saml.core.domain.IdaStatus;
+
+public interface AuthenticationStatusFactory<T extends Enum, U extends IdaStatus> {
```

Comment:
Extends Enum is a bit generic. I'm pretty sure we want something more specific to our own codebase, don't we?

Reply:
The status codes extend an Enum so there's no much else 'T' can extend.... We don't have a class or anything. We _could_ create one and it would be nice but that's a story I'd play separately